codechecker_common/configuration_access.py

```diff
@@ -0,0 +1,1208 @@
+                description="<Root of Schema>")
+        })
+
+    @property
+    def options(self) -> OptionDict:
+        return object.__getattribute__(self, _K_OPTIONS)
+
+    @property
+    def root(self) -> OptionDirectory:
+        return cast(OptionDirectory, self.options['/'])
+
+    def add_option(self,
+                   name: Optional[str],
+                   path: str,
+                   parent: Optional[OptionDirectory] = None,
+                   **kwargs) -> OptionBase:
+        """
+        Registers an `Option` in the current `Schema`.
+
+        The apparent path of the to-be-created `Option` determines its type:
+          - paths ending in ``"[]/"`` denote an `OptionDirectoryList`, which
+            is a numbered list of `OptionDirectory`s, containing multiple
+            instances of `Option`s.
+          - paths ending in ``'/'`` denote an `OptionDirectory`.
+          - everything else denotes an `Option`, which is a leaf value.
+
+        If path begins with ``"./"``, the `parent` parameter should be set,
+        and path is understood relative to the parent.
+        By default, the `parent` is the same as the `Root` of the schema.
+
+        Additional keyword arguments are forwarded to the `Option` constructor.
+        """
+        if path == '/':
+            raise ValueError("The '/' root of the Option structure is "
+                             "hard-coded and can not be manually added as "
+                             "an option!")
+        if not path.startswith(('/', "./")):
+            raise ValueError(
+                f"Path '{path}' must be a proper relative or absolute "
+                "POSIX-y path")
+        if path.endswith("[]"):
+            raise ValueError(
+                f"Path '{path}' designating an indexable sequence must use "
+                "the directory syntax, and end with \"[]/\"")
+        if ' ' in path:
+            raise ValueError(f"Path '{path}' must not contain spaces")
+
+        clazz: Type[OptionBase] = Option
+        if path.endswith('/'):
+            clazz = OptionDirectory
+            path = path.rstrip('/')
+            if path.endswith("[]"):
+                clazz = OptionDirectoryList
+
+        if path.startswith("./"):
+            if parent is None:
+                parent = self.root
+            path = parent._path.rstrip('/') + '/' + path.replace("./", '', 1)
+
+        options = object.__getattribute__(self, _K_OPTIONS)
+        for parent_path in map(str, reversed(PurePosixPath(path).parents)):
+            if parent_path not in options:
+                raise KeyError(f"Parent OptionDirectory-like '{parent_path}' "
+                               "is not registered")
+
+        opt = clazz(self, name, path, **kwargs)
+        if opt._path in options:
+            raise KeyError(f"Option '{opt._path}' is already registered!")
+        options[opt._path] = opt
+        return opt
+
+
+def _get_config_json(file_path: Path) -> Dict[str, Any]:
```

Comment:
The only way it would make sense there as if it was a `@classmethod` or even better, a `@staticmethod`. However, this function does not consider anything related to the nice schematic access of configurations, so it would look odd to have it in there, a function that "talks" on a completely different abstraction level.